data/hai missions.txt

```diff
@@ -13,6 +13,7 @@ mission "First Contact: Hai"
 	source
 		attributes "hai"
 	on offer
+		log "Made contact with a species north of human space, through a wormhole. They call themselves the Hai, and seem to be peaceful. But they warn of renegades in the north of their space."
```

Comment:
You could put in a check to see if the player has been through the wormhole. OK, you could have been through the wormhole and not seen a Hai ship but I think that's unlikely.